ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -481,8 +481,8 @@ func (r *RayClusterReconciler) reconcileIngress(ctx context.Context, instance *r
 func (r *RayClusterReconciler) reconcileRouteOpenShift(ctx context.Context, instance *rayv1.RayCluster) error {
 	logger := ctrl.LoggerFrom(ctx)
 	headRoutes := routev1.RouteList{}
-	filterLabels := client.MatchingLabels{utils.RayClusterLabelKey: instance.Name}
-	if err := r.List(ctx, &headRoutes, client.InNamespace(instance.Namespace), filterLabels); err != nil {
+	filterLabels := common.RayClusterAllPodsAssociationOptions(instance).ToListOptions()
```

Comment:
It's misleading to use `RayClusterAllPodsAssociationOptions` to associate K8s ingress / route. Please create a new function `RayClusterNetworkResourcesOptions` for both ingress and route instead.